### ISSUE TITLE
Cast call results in "Ambigous method call" (again)

### ISSUE THREAD

**Issue by disarray2077** [beefytech/Beef#1416]
Basically, this is the same issue from https://github.com/beefytech/Beef/issues/1403, but now it only gives an error if I add a `ReadOnlySpan` conversion operator to the type.

Error:
```
ERROR: Ambiguous method call
  while compiling 'test6.Program'
  > 'test6.Program.Test.operator test6.Program.ReadOnlySpan<int>(test6.Program.Test self)' is a candidate at line 234:36 in d:\BeefLang\test6\src\Program.bf
    [Inline] public static implicit operator ReadOnlySpan<int>(Self self)
                                    ^^^^^^^^
  > 'test6.Program.Test.operator System.Span<int>(test6.Program.Test self)' is a candidate at line 237:36 in d:\BeefLang\test6\src\Program.bf
    [Inline] public static implicit operator Span<int>(Self self)
                                    ^^^^^^^^
```

Code:
```
static
{
	[Inline]
	public static void ForEach<TList, TValueType, TDlg>(this TList list, TDlg dlg)
		where TList : concrete, IEnumerable<TValueType>
		where TDlg : delegate void(TValueType)
	{
		for (let item in list)
			dlg(item);
	}
}

namespace test6
{
	class Program
	{
		struct ReadOnlySpan<T> : IEnumerable<T>
		{
			public IEnumerator<T> GetEnumerator()
			{
				return default;
			}
		}

		struct Test : IEnumerable<int>
		{
			public static implicit operator List<int>(Self self)
				=> default;

			public static implicit operator ReadOnlySpan<int>(Self self)
				=> default;

			public static implicit operator Span<int>(Self self)
				=> default;

			public IEnumerator<int> GetEnumerator()
				=> default;
		}

		public static int Main(String[] args)
		{
			Test test = .();
			test.ForEach((i) => NOP!());

			return 0;
		}
	}
}
```

**Comment by bfiete** [comment]
Ah- that's because it is exactly the same bug. I "temporarily" reverted the fix while investigating a different bug and then failed to un-revert the fix. I'll commit the re-fix tomorrow.

**Comment by bfiete** [comment]
Re-fixed at 6d4e09cf1373979bbe790ce29de59502e3456888
